### DISCUSSION TITLE
How to use on_conflict_do_nothing with SQLite database

### DISCUSSION THREAD

**Discussion by ikus060** [sqlalchemy/sqlalchemy#7007] [Usage Questions] [answered]
I'm trying to use the on_conflict_do_nothing() on insert statement. But I get `Insert' object has no attribute 'on_conflict_do_nothing'`

Seams to be realated to the fact, the the Insert object is not an SQLite Insert object.

Do I need to do something specific to get `on_conflict_do_nothing` working with Sqlite.

Here a snippet:
```
from sqlalchemy import Column, MetaData, String, Table, create_engine

META = MetaData()

MYTABLE = Table(
    'mytable', META,
    Column('col1', String, primary_key=True),
    Column('col2', String),
    sqlite_autoincrement=True,
)

db = create_engine('sqlite:////tmp/test.db')
META.create_all(db)

with db.connect() as conn:
    conn.execute(MYTABLE.insert().values(
        col1='val1', col2='val2').on_conflict_do_nothing())

```

**Comment by CaselIT** [comment, marked as answer]
Hi,

You need to use the insert from the `sqlalchemy.dialect.sqlite` module, as mentioned here https://docs.sqlalchemy.org/en/14/dialects/sqlite.html?highlight=insert#sqlalchemy.dialects.sqlite.insert

Table.insert returns a general insert that does not have dialect specific methods
